mypy/checker.py

```diff
@@ -2484,7 +2484,14 @@ class C(B, A[int]): ...  # this is unsafe because...
         first_type = get_proper_type(self.determine_type_of_member(first))
         second_type = get_proper_type(self.determine_type_of_member(second))
 
-        if isinstance(first_type, FunctionLike) and isinstance(second_type, FunctionLike):
+        # start with the special case that Instance can be a subtype of FunctionLike
+        call = None
+        if isinstance(first_type, Instance):
+            call = find_member("__call__", first_type, first_type, is_operator=True)
+        if call and isinstance(second_type, FunctionLike):
+            second_sig = self.bind_and_map_method(second, second_type, ctx, base2)
+            ok = is_subtype(call, second_sig, ignore_pos_arg_names=True)
```

Comment:
Yes, there are possibly more places where similar checks are performed (or should be performed). 

A quick search for `find_member("__call__",`: five hits that seem to serve the same or similar purposes besides my own one.

However, there seem to be many small differences.  So, refactoring it might not be simple.

Reply:
I had a second look, just searching for `"__call__"`.  51 hits.  Many special cases.  I have no idea how far a generalisation like the one you suggested would be helpful.  And it would be too much effort for me, currently.

Reply:
Yeah, totally understood. BTW I'm just trying to help with reviews, I'm not a maintainer.

Reply:
I see.  Do you know how to proceed?  Besides this pull request (which I really would like to land in Mypy 1.2), I have two older ones open (#14440 and  #14390).  I'm not sure who to ping when and how...